hasher-matcher-actioner/scripts/script_utils.py

```diff
@@ -30,14 +62,38 @@ def _get_header(self) -> t.Dict[str, str]:
     def _get_request_url(self, api_path: str) -> str:
         return urljoin(self.api_url, api_path)
 
+    def _get_with_auth(
```

Comment:
A more flexible way to do this is to expose a `requests.Session` object [1]. That way users have access to the full gamut of requests capabilities while re-using the auth. A session object has almost all `request.*` methods available.

Imagine having to write `_post_multi_part_with_auth()` and `_put_with_auth()` and `_delete_with_auth()`

[1]: https://docs.python-requests.org/en/master/user/advanced/

Reply:
Oh, and did I mention it respects Keep-Alive and re-uses the underlying TCP connection for higher performance?

Reply:
sure! I was avoiding cookies but since we control the API and don't set any the concern is likely unfounded.